src/modules/dashboard/components/Dialogs/CancelExpenditureDialog/CancelExpenditureForm.tsx

```diff
@@ -0,0 +1,261 @@
+import React, { useCallback, useState } from 'react';
+import classNames from 'classnames';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import { FormikProps } from 'formik';
+
+import Dialog, { DialogSection } from '~core/Dialog';
+import { tokens as tokensMock } from '~dashboard/ExpenditurePage/ExpenditureSettings/constants';
+import { Annotations, Radio, Toggle } from '~core/Fields';
+import { Colony, useLoggedInUser } from '~data/index';
+import Heading from '~core/Heading';
+import TokenIcon from '~dashboard/HookedTokenIcon';
+import Button from '~core/Button';
+import Numeral from '~core/Numeral';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+import { useTransformer } from '~utils/hooks';
+import { getAllUserRoles } from '~modules/transformers';
+import { hasRoot } from '~modules/users/checks';
+import { useDialogActionPermissions } from '~utils/hooks/useDialogActionPermissions';
+import MotionDomainSelect from '~dashboard/MotionDomainSelect';
+
+import { PenalizeType } from './types';
+import { FormValues } from './CancelExpenditureDialog';
+import styles from './CancelExpenditureDialog.css';
+import QuestionMarkTooltip from '~core/QuestionMarkTooltip';
+
+const MSG = defineMessages({
+  header: {
+    id: 'dashboard.CancelExpenditureDialog.CancelExpenditureForm.header',
+    defaultMessage: 'Cancel payment',
+  },
+  ownersStake: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.ownersStake`,
+    defaultMessage: `Owner's stake`,
+  },
+  shouldBePenalized: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.shouldBePenalized`,
+    defaultMessage: 'Do you want to penalize the owner?',
+  },
+  penalize: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.penalize`,
+    defaultMessage: 'Penalize',
+  },
+  showMercy: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.showMercy`,
+    defaultMessage: 'Show mercy',
+  },
+  penalizeMessage: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.penalizeMessage`,
+    defaultMessage: 'Owner will lose their stake and equivalent reputation.',
+  },
+  mercyMessage: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.mercyMessage`,
+    defaultMessage: 'Owner will keep their stake and reputation.',
+  },
+  submit: {
+    id: 'dashboard.CancelExpenditureDialog.CancelExpenditureForm.submit',
+    defaultMessage: 'Submit',
+  },
+  textareaLabel: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.textareaLabel`,
+    defaultMessage: `Explain why you're cancelling this payment (optional)`,
+  },
+  effectTooltip: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.effectTooltip`,
+    defaultMessage: `Decide what to do with the owner's stake when cancelling this Advanced Payment.`,
+  },
+  createDomain: {
+    id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm.creationTarget`,
```

Comment:
The final part of this id should match the message label createDomain: 
id: `dashboard.CancelExpenditureDialog.CancelExpenditureForm. createDomain `,